src/main/java/teammates/ui/controller/AdminActivityLogPageAction.java

```diff
@@ -225,64 +228,49 @@ private LogQuery buildQuery(AdminActivityLogPageData data) {
-        }       
+        if (isVersionSpecifiedInRequest) {
+            return versions;
+        }
         return getDefaultVersionIdsForQuery();
     }
     
+    /**
+     * Gets a list of versions, including the current version and 5 preceding versions (if available).
+     * @return a list of default versions for query.
+     */
     private List<String> getDefaultVersionIdsForQuery() {
-    
-        String currentVersion = Config.inst().getAppVersion();
         List<String> defaultVersions = new ArrayList<String>();
         
-        //Check whether version Id contains alphabet 
-        //Eg. 5.05rc
-        if (currentVersion.matches(".*[A-z.*]")) {
-            //if current version contains alphatet,
-            //by default just prepare current version as a single element for the query
-            defaultVersions.add(currentVersion.replace(".", "-"));
-            
-        } else {
-            //current version does not contain alphabet
-            //by default prepare current version with preceding 3 versions
-            defaultVersions = getRecentVersionIdsWithDigitOnly(currentVersion);
-        }
-        
-        return defaultVersions;        
-    }
-    
-    private List<String> getRecentVersionIdsWithDigitOnly(String currentVersion) {
-        
-        List<String> recentVersions = new ArrayList<String>();
-        
-        double curVersionAsDouble = Double.parseDouble(currentVersion);
-        recentVersions.add(currentVersion.replace(".", "-"));
-        
-        //go back for three preceding versions
-        //subtract from double form of current version id
-        //Eg. current version is 4.01 --> 4.00, 3.99, 3.98  --> 4-00, 3-99, 3-98
-        for (int i = 1; i < 4; i++) {
-
-            double preVersionAsDouble = curVersionAsDouble - 0.01 * i;
-            if (preVersionAsDouble > 0) {
-                String preVersion = String.format("%.2f", preVersionAsDouble)
-                                          .replace(".", "-");
-                
-                recentVersions.add(preVersion);
+        ModulesService modulesService = ModulesServiceFactory.getModulesService();
+        Set<String> versionList = modulesService.getVersions(null); // null == default module
+        String currentVersion = modulesService.getCurrentVersion();
+        defaultVersions.add(currentVersion);
+        boolean isCurrentVersionFound = false;
+        
+        // Find the current version then get at most 5 versions below it.
+        for(String version : versionList) {
+            if (version.equals(currentVersion)) {
+                isCurrentVersionFound = true;
+            } else {
+                if (isCurrentVersionFound) {
+                    if (defaultVersions.size() < MAX_DEFAULT_VERSION_NUMBER) {
+                        defaultVersions.add(version);
+                    } else {
+                        return defaultVersions;
+                    }
+                }
```

Comment:
This `for` look looks like it has too many nesting levels for such a short code block. See if you can simplify.